apps/meteor/client/views/invite/hooks/useValidateInviteQuery.ts

```diff
@@ -0,0 +1,53 @@
+import { useEndpoint, useSessionDispatch, useSetting, useToastMessageDispatch, useTranslation } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+
+import { useInviteTokenMutation } from './useInviteTokenMutation';
+
+export const useValidateInviteQuery = (userId: string, token: string) => {
+	const t = useTranslation();
+
+	const registrationForm = useSetting('Accounts_RegistrationForm');
+
+	const setLoginDefaultState = useSessionDispatch('loginDefaultState');
+	const dispatchToastMessage = useToastMessageDispatch();
+	const mutateInviteToken = useInviteTokenMutation();
+
+	const handleValidateInviteToken = useEndpoint('POST', '/v1/validateInviteToken');
+
+	return useQuery(
+		['invite', token],
+		async () => {
+			if (!token) {
+				return false;
+			}
+
+			try {
+				const { valid } = await handleValidateInviteToken({ token });
+
+				return valid;
+			} catch (error) {
+				dispatchToastMessage({ type: 'error', message: t('Failed_to_validate_invite_token') });
+				return false;
+			}
+		},
+		{
+			onSuccess: async (valid) => {
+				if (!token) {
+					return;
+				}
+
+				if (registrationForm !== 'Disabled') {
+					setLoginDefaultState('register');
+				} else {
+					setLoginDefaultState('login');
+				}
+
+				if (!valid || !userId) {
```

Comment:
If I understood this flow correctly the mutation should only run after the user logged in, otherwise, the rerouting would occur at the wrong moment and consequently break the process, the userid verification confirms the user has logged in and allows the mutation to run which on success will also run the rerouting.

Reply:
Understood... All good then